src/emr-serverless/commands/startApplication.ts

```diff
@@ -0,0 +1,54 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { getLogger } from '../../shared/logger'
+import { EmrServerlessApplicationNode } from '../explorer/emrServerlessApplicationNode'
+import { Commands } from '../../shared/vscode/commands'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { showViewLogsMessage } from '../../shared/utilities/messages'
+
+export async function startApplication(
+    node: EmrServerlessApplicationNode,
+    commands = Commands.vscode()
+): Promise<void> {
+    getLogger().debug('EMR Serverless: StartApplication called for %O', node)
+
+    const applicationId = node.application.id
+
+    await startWithProgress(node)
+        .catch(e => {
+            getLogger().error(`Failed to start application ${applicationId}: %s`, e)
+            showViewLogsMessage(
+                localize(
+                    'AWS.emrserverless.startApplication.failure',
+                    'Failed to start application: {0}',
+                    applicationId
+                )
+            )
+        })
+        .finally(() => commands.execute('aws.refreshAwsExplorerNode', node.parent))
+    getLogger().info(`EMR Serverless: started application: ${applicationId}`)
+
+    // vscode.window.showInformationMessage(
+    //     localize('AWS.emrserverless.startApplication.success', 'Started application: {0}', applicationId)
+    // )
+}
+
+async function startWithProgress(node: EmrServerlessApplicationNode): Promise<void> {
```

Comment:
showProgressWithTimeout may be useful. https://github.com/aws/aws-toolkit-vscode/blob/8428a21785c0d74b1a2f660d5b2a8347509e25b4/src/shared/utilities/messages.ts#L157

Reply:
Indeed it may! TY.